### ISSUE TITLE
Discriminator field does not change upon selecting different models in request sample

### ISSUE THREAD

**Issue by debri174** [Rebilly/ReDoc#524]
I'm running into a bug with the discriminator field in Request Samples. When I click on the various models the value in the discriminator field does not change. 

Here is a portion of my spec that shows the discriminator setup:

```
pageelementscommon:
    type: object
    discriminator: type
    properties:
      id:
        type: integer
        description: Question ID
        readOnly: true
      type:
        type: string
        description: Element type used to determine schema to post/returned
        enum:
        - cascadingdropdown
        - checkboxes
        - contactform
        - contsum
        - customgroup
        - customtable
        - date
        - dropdown
        - email
        - essay
        - fileupload
        - gridcheckboxes
        - griddropdown
        - gridradiobutton
        - gridstars
        - gridtextbox
        - imagemulti
        - imagesingle
        - instructions
        - likert
        - listdropdown
        - listtextbox
        - nps
        - number
        - percent
        - radiobuttons
        - rankdragdrop
        - rankgrid
        - signature
        - slider
        - slidermulti
        - textbox
        - customscriptaction
        - hiddenvalueaction
        - javascriptaction
        - loginaction
        - pagetimeraction
        - percentbranchaction
        - quizscoreaction
        - recaptchaaction
        - resultschartaction
        - reviewaction
        - sectionnavaction
        - emailaction
        - socialmediaaction
        - urlredirectaction
        - webhookaction
        - video
        - audio
```

Here's a demo of the issue:

![request-sample-discrim-bug](https://user-images.githubusercontent.com/32558006/40890135-6a4e1166-672e-11e8-9787-15ddec75f689.gif)

This is problematic because, as you can see, the subsequent fields are often the same for the various models so the end user might think that the examples are not changing even though they are. 

Thanks for such a great tool!

**Comment by debri174** [comment]
A related question/suggestion: I am using the CLI do deploy the docs and note that it is converting the spec to v3.0. Is there support for v3 polymorphism in Redoc? Meaning, will it apply the dropdowns to various models as they are specified in v3?

**Comment by debri174** [comment]
I tested the above an confirmed that polymorphism of as specified in OAS 3.0 displays quite nicely! Kudos! The original issue remains though:
![v3-examples](https://user-images.githubusercontent.com/32558006/41005073-465ef1ce-68d9-11e8-89f7-cf481a3d45b9.gif)

**Comment by trip-somers** [comment]
@RomanGotsiy Should this be closed? @debri174 said, "The original issue remains."

Also, @debri174, how did you get it to display your samples like that? I've attempted a number of configurations, and I've come up empty.

**Comment by RomanGotsiy** [comment]
@trip-somers I think yes yes. The issue was closed after @debri174 comment in 5dae267.
